docs/DEVELOPING.md

```diff
@@ -343,6 +343,139 @@ $ make ftest NAME=azure_blob_storage
+
+  - `table_1, table_2`
+  - `*`
+
+##### `fetch_size`
+
+Number of rows to fetch on each request to Oracle Database. Default value is `50`.
+
+##### `retry_count`
+
+The number of retry attempts after failed request to Oracle Database. Default value is `3`.
+
+##### `oracle_protocol`
+
+The protocol which the connector uses to establish a connection. Default value is `TCP`. Note: For a secured connection, a user needs to use `TCPS`.
+
+##### `oracle_home`
+
+Path of the Oracle home directory to run connector with thick mode for secured connection. Note: In case of unsecured connection, keep `oracle_home` field empty.
+
+##### `wallet_configuration_path`
+
+Path of the oracle configuration files. Only applicable when configuration files are not at the default location. Note: If Wallet configuration files are located at default location, keep `wallet_configuration_path` field empty.
+
+### Setup for a secured connection with Oracle
+ - User needs to install the Oracle service on the system where the connector is running.
+ - Set the `oracle_home` parameter to the location of your Oracle home directory. If configuration files are not at the default location, set `wallet_configuration_path` parameter.
+  - Now create a directory to store the wallet.
+```shell
+$ mkdir $ORACLE_HOME/ssl_wallet
+```
+  - User needs to create `sqlnet.ora` at `$ORACLE_HOME/network/admin` location and add the following content.
+```
+WALLET_LOCATION = (SOURCE = (METHOD = FILE) (METHOD_DATA = (DIRECTORY = $ORACLE_HOME/ssl_wallet))) 
+SSL_CLIENT_AUTHENTICATION = FALSE 
+SSL_VERSION = 1.0 
+SSL_CIPHER_SUITES = (SSL_RSA_WITH_AES_256_CBC_SHA) 
+SSL_SERVER_DN_MATCH = ON
+```
+  - Then execute the following commands to create a wallet and attach an SSL certificate.
+```shell
+$ orapki wallet create -wallet path-to-oracle-home/ssl_wallet -auto_login_only
+
+$ orapki wallet add -wallet path-to-oracle-home/ssl_wallet -trusted_cert -cert path-to-oracle-home/ssl_wallet/root_ca.pem -auto_login_only
+```
+Replace the file name with your file.
+
+ℹ️ For more information, please refer following [documentation](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.Oracle.Options.SSL.html)
+
+
+### Connector Limitations
+
+- If the value of the table's system change number (SCN) is not between the `min(SCN)` and `max(SCN)` values of the `SMON_SCN_TIME` table, the connector will not be able to retrieve the most recently updated time, and data will index in every sync. For more details check out the following [documentation](https://community.oracle.com/tech/apps-infra/discussion/4076446/show-error-about-ora-08181-specified-number-is-not-a-valid-system-change-number-when-using-scn-t).
+- The Oracle Database connector does not allow the `sys` user as it contains 1000+ system tables which are irrelevant for users. If the user is working with `sys` user, the user needs to pass either `sysdba` or `sysoper` mode.
```

Comment:
For users who are fetching from non-system tables, can we specify the minimum privileges required for the database user for the connector to fetch data from these user tables?